### ISSUE TITLE
[$250] Approval flow - Non-admin approver does not have access to workspace chat with submitter

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53937]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.74-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5324457
**Email or phone of affected tester (no customers):** applausetester+kh1311020@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Admin invites User B and C (both are members) to the workspace.
- Workflows are enabled.
- Add approvals are enabled.
- User B is the approver, but not an admin.

1. Go to staging.new.expensify.com
2. [User C] Go to workspace chat.
3. [User C] Submit an expense.
4. [User B] Open the report as approver.

## Expected Result:
User B (non-admin approver) should have access to workspace chat with User C.

## Actual Result:
User B (non-admin approver) does not have access to workspace chat with User C. User B can only access the expense report and not the main chat.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/0e6f0190-c1a4-42cf-a60d-3d78a4341300

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866909364662980021</li>
        <li>Upwork Job ID: 1866909364662980021</li>
        <li>Last Price Increase: 2025-01-01</li>
    </ul>
</details>

**Comment by Sethrollinsbah** [comment]
Contributor details
Your Expensify account email: sethryanrollins@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b495158df32e9c89

**Comment by dixitparkhiya** [comment]
Contributor details
Your Expensify account email: [parkhiyadixit@gmail.com](parkhiyadixit@gmail.com)
Upwork Profile Link: [https://www.upwork.com/freelancers/~01a2817e066c68c00f](https://www.upwork.com/freelancers/~01a2817e066c68c00f)

**Comment by narefyev91** [comment]
Hi, I'm Nicolay from Callstack - expert contributor group - and I would like to work on this issue.

**Comment by beodw** [comment]
Hi, tried reproducing but there seems to be a bug with adding members to the workspace. I one user only see 2 members in the workspace while the other sees all 3.

https://github.com/user-attachments/assets/6f2d56ff-d313-4bb8-a2e6-400be4cd27e0

**Comment by narefyev91** [comment]
Continue investigation possible root causes

**Comment by narefyev91** [comment]
@IuliiaHerets @greg-schroeder @mollfpr - was not able to reproduce on latest dev env:

https://github.com/user-attachments/assets/fdba9880-0be3-4662-b762-b23b5007e7a4

Based on the code - it seems like issue was fixed with a new way of identify getReport function (https://github.com/Expensify/App/pull/53385) - which was merged not so far ago

**Comment by greg-schroeder** [comment]
Thanks @narefyev91 - I suppose we close then!

**Comment by IuliiaHerets** [comment]
QA team still can repro the issue in the latest build 

https://github.com/user-attachments/assets/88903974-eb02-4999-8d57-6f065f188f97

**Comment by IuliiaHerets** [comment]
@narefyev91 Done
[1812.txt](https://github.com/user-attachments/files/18177751/1812.txt)

**Comment by mollfpr** [comment]
@narefyev91 I agree with you, this is a BE issue. I still unable to reproduce the issue.

**Comment by narefyev91** [comment]
> @narefyev91 I agree with you, this is a BE issue. I still unable to reproduce the issue.

I was able to do it - (based on the videos) - most important - that user (who should approve expense) - should be already logged in. In that case user will not receive any information about new chat

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mollfpr** [comment]
I still can't reproduce the issue and after assigning the user B for approver role, all the chats from other members appear on user B.

**Comment by narefyev91** [comment]
@mollfpr could we ask BE engineer to get into this one?

**Comment by mollfpr** [comment]
@greg-schroeder Could you add the Internal label? Thank you!

**Comment by greg-schroeder** [comment]
Will do!

**Comment by dubielzyk-expensify** [comment]
Assume I was assigned by mistake with the design label, so I'll unassign, but let me know if you need any visual eyes on it 👍

**Comment by greg-schroeder** [comment]
Oops, yes, sorry about that @dubielzyk-expensify

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by greg-schroeder** [comment]
This might have been resolved - not able to reproduce

**Comment by greg-schroeder** [comment]
Closing as resolved elsewhere; please comment/reopen if you disagree.
